src/components/RewardsModal/RewardsModal.tsx

```diff
@@ -0,0 +1,72 @@
+import { Button, ButtonColor } from '../Button/Button'
+import { shortenAddress } from '../../utils/shortenAddress'
+import { Modal } from '../Modal/Modal'
+import { format } from '../../utils'
+import { useEffect, useState } from 'react'
+import { getConnection } from '../../utils/useConnect'
+import { femtoKiltToDigits } from '../../utils/conversion'
+
+export interface Props {
+  accountAddress: string
+  rewards: number
+  onConfirm: () => void
+  closeModal: () => void
+}
+
+export const RewardModal: React.FC<Props> = ({
+  accountAddress,
+  rewards,
+  onConfirm,
+  closeModal,
+}) => {
+  const shortAddress = shortenAddress(accountAddress)
+  const [fee, setFee] = useState<number>(0.0003)
+
+  useEffect(() => {
+    const getFee = async () => {
+      const api = await getConnection()
+      const increment = await api.tx.parachainStaking.incrementDelegatorRewards().paymentInfo(accountAddress)
+      const claim = await api.tx.parachainStaking.claimRewards().paymentInfo(accountAddress)
+
+      if (increment.partialFee || claim.partialFee) {
+        const feeInFemto = (increment.partialFee.toBigInt() || 0n) + (claim.partialFee.toBigInt() || 0n)
```

Comment:
Did you verify the cost? I think what you may actually want is
```suggestion
const increment = api.tx.parachainStaking.incrementDelegatorRewards()
const claim = api.tx.parachainStaking.claimRewards()
const batch = api.tx.utility.batch([increment, claim])
const feeInFemto = (await batch.paymentInfo(accountAddress)).partialFee.toBigInt()
```
The result is different from the sum of both calls, probably because there is a base fee for signature verification.
Sum in femto: 214000006232
Empty batch in femto: 108000003116
Batch of both in femto: 112000003116